tests/test.cpp

```diff
@@ -4179,6 +4179,47 @@ void FieldIdentifierTest() {
 #endif
 }
 
+void NestedVerifierTest() {
+  // Create a nested monster.
+  flatbuffers::FlatBufferBuilder nested_builder;
+  MonsterT nested_monster;
+  FinishMonsterBuffer(nested_builder,
+                      CreateMonster(nested_builder, &nested_monster));
+
+  // Copy the complete nested monster to a vector
+  const std::vector<uint8_t> nested_monster_bytes(
+      nested_builder.GetBufferPointer(),
+      nested_builder.GetBufferPointer() + nested_builder.GetSize());
+
+  // Create the root monster.
+  flatbuffers::FlatBufferBuilder builder;
+  MonsterT monster;
+
+  // Set the nested monster bytes.
+  monster.testnestedflatbuffer = std::move(nested_monster_bytes);
```

Comment:
I guess this is nice to test the Object API, but this really makes 2 (!) unnecessary copies (first the nested_monster_bytes above, then the copy from the MonsterT into the real flatbuffer.. if you hadn't used `std::move` here it would be 3 copies).
That shouldn't matter for a test, but really it is good to use "idiomatic" (which for flatbuffers means "efficient") code where possible. Not using the object API, you could serialize the bytes pointer to by `GetBufferPointer` straight into the parent buffer without extra copies.

Reply:
Yeah, I was focused on what I was testing more than the efficiency. I was aiming for conceptual efficiency more than runtime. 

If you want me to change it to using the standard API, I can.

Reply:
This got me thinking too, I didn't realize that testnestedflatbuffer only took a `std::vector<uint8_t>`, it would be nice to take in a `MonsterT *`  directly and avoid the copy.

Reply:
I don't think it be much more using the standard API.

A nested FlatBuffer taking a `MonsterT *` wouldn't avoid a copy, since it has to first serialize that to bytes before it can add that. We'd could avoid the copy if we serialized the inner one with a special allocator that knows how to use the outer one as a buffer, assuming alignment works out etc.. but that doesn't make sense to do this for the Object API since that is already a copy and is already less efficient. If we wanted to have this kind of in-place nested construction, we should make that for the base API, and then the Object API could follow.

Reply:
Switched to the base API. Its only annoying since the `name` field is required, so you have to do a bit of extra work to make valid.